dali/test/python/test_operator_nonsilence.py

```diff
@@ -0,0 +1,120 @@
+                                     frame_length=frame_length,
+                                     hop_length=hop_length)
+
+    # librosa's trim function calculates power with reference to center of window,
+    # while DALI uses beginning of window. Hence the subtraction below
+    begin = index[0] - frame_length / 2
+    length = index[1] - index[0]
+    return np.array([begin]), np.array([length])
+
+
+class NonsilencePipeline(Pipeline):
+    def __init__(self, batch_size, num_threads=1, exec_async=True, exec_pipelined=True):
+        super(NonsilencePipeline, self).__init__(batch_size, num_threads, 0, seed=42,
+                                                 exec_async=exec_async,
+                                                 exec_pipelined=exec_pipelined)
+        self.input = ops.FileReader(device="cpu", file_root=audio_files)
+        self.decode = ops.AudioDecoder(device="cpu", dtype=types.FLOAT, downmix=True)
+
+        self.nonsilence = None
+
+    def define_graph(self):
+        if self.nonsilence is None:
+            raise RuntimeError(
+                "Error: you need to derive from this class and define `self.nonsilence` operator")
+        read, _ = self.input()
+        audio, rate = self.decode(read)
+        begin, len = self.nonsilence(audio)
+        return begin, len
+
+
+class NonsilenceDaliPipeline(NonsilencePipeline):
+    def __init__(self, batch_size, cutoff_value, window_size, reference_db, reference_max,
+                 reset_interval):
+        super(NonsilenceDaliPipeline, self).__init__(batch_size, num_threads=1)
+        self.nonsilence = ops.NonsilenceRegion(cutoff_db=cutoff_value, window_length=window_size,
+                                               reference_db=reference_db,
+                                               reference_max=reference_max,
+                                               reset_interval=reset_interval,
+                                               device="cpu")
+
+
+class NonsilenceRosaPipeline(NonsilencePipeline):
+    def __init__(self, batch_size, cutoff_value, window_size, reference_db, reference_max,
+                 reset_interval):
+        super(NonsilenceRosaPipeline, self).__init__(batch_size, num_threads=1,
+                                                     exec_async=False, exec_pipelined=False)
+        hop_length = 1
+        function = partial(trim_ref, cutoff_value, np.max if reference_max else reference_db,
+                           window_size, hop_length)
+        self.nonsilence = ops.PythonFunction(function=function, num_outputs=2)
+
+
+def check_nonsilence_operator(batch_size, cutoff_value, window_size, reference_db, reference_max,
+                              reset_interval, eps):
+    dali_pipe = NonsilenceDaliPipeline(batch_size, cutoff_value, window_size, reference_db,
+                                       reference_max, reset_interval)
+    rosa_pipe = NonsilenceRosaPipeline(batch_size, cutoff_value, window_size, reference_db,
```

Comment:
Can you use compare_pipelines?

Reply:
Unfortunately no, it's explained in the comment slightly below. 
> This test shall pass either when the lengths match and are equal to 0 or when both lengths and begins match 

This happens, because when NonsilenceRegion's `length == 0`, `begin` is undefined. `compare_pipelines` forces all outputs to match (or at least I think so).

Reply:
why is begin undefined? Can't we just have 0 or something?

Reply:
Seems natural to me, that when something is of length 0, it doesn't have begin index. If any, I'd go for `-1` or something, that's definately not an index (`0` is...)

Reply:
`-1` sounds like a good choice,